packages/form-js-viewer/src/render/hooks/useReadonly.js

```diff
@@ -0,0 +1,32 @@
+import useService from './useService.js';
+import useFilteredFormData from './useFilteredFormData.js';
+
+/**
+ * Retrieve readonly value of a form field, given it can be an
+ * expression optionally or configured globally.
+ *
+ * @typedef { import('../../types').FormProperties } FormProperties
+ *
+ * @param {any} formField
+ * @param {FormProperties} properties
+ *
+ * @returns {boolean}
+ */
+export default function useReadonly(formField, properties = {}) {
+  const expressionLanguage = useService('expressionLanguage');
+  const conditionChecker = useService('conditionChecker', false);
+  const filteredData = useFilteredFormData();
+
+  const { readonly } = formField;
+
+  if (properties.readOnly) {
+    return true;
+  }
+
+  // todo(pinussilvestrus): memoize?
```

Comment:
Can we remove or create an issue for this?

Reply:
I removed it.